embed.fnc

```diff
@@ -130,6 +130,47 @@
 :         yourself, and from within it, refer to the function in its full
 :         'Perl_' form with any necessary thread context parameter.
 :
+: AUTOMATIC PARAMETER SANITY CHECKING
+:
+:   regen/embed.pl generates macros for each entry (unless excluded by the -G
+:   flag) that do a bit of parameter validation.  If your element is named
+:   'foo()', the generated macro will be named 'PERL_ARGS_ASSERT_FOO'.  You
+:   should place a call to that macro before any other code in foo().  It will
+:   automatically expand to whatever checking is currently generated for foo
+:   (often none).  These are in the form of assert() calls, so only are
+:   activated for DEBUGGING builds.
+:
+:   You must specify what checking is needed for all pointer arguments.  If the
+:   pointer is allowed to point to NULL, prefix that argument with 'NULLOK'
+:   (following the template of the many entries in this file that have that).
+:   If it can't be NULL, use 'NN' (again many entries herein do that).
+:   The reason for this requirement is to tell the maintainers that you have
+:   considered the question about the argument, and this is the answer.
+:
+:   For a numeric argument, you may specify that it can't be 0 by using 'NZ'
+:
+:   regen/embed.pl may automatically add further checking for any argument as
+:   it deems desirable.  You can override this by specifying 'NOCHECK'
+:
+:	Currently this further checking is just for pointer parameters that
+:	point to AVs, CVs or HVs.  The check is that the SV being pointed to is
+:	of the intended type, by inspecting its SvTYPE(). For some functions
+:	this check may be inappropriate, as in rare cases the arguments passed
+:	may not be of the correct type. As already mentioned, NOCHECK
+:	suppresses this check.
+:
+:   Currently, it is optional to include an empty ARGS_ASSERT macro in your
+:   functions.  But a porting test enforces that a non-empty one be included.
+:   This should be at the top of your function so that the sanity checks have
+:   passed before anything tries to use an argument.  When writing a new
+:   function, add the macro even if not required, and you'll never have to go
+:   back and add one later when more checks get added
+:
+:   The G flag does not suppress the generation of a non-empty ARGS_ASSERT
+:   macro
```

Comment:
Can we state this without a triple negative? ("does not", "suppress", "non-empty")

Maybe:
```
The G flag only suppresses empty ARGS_ASSERT macros.
```
though I'm not sure we need anything in addition to the G flag documentation below.

Reply:
removal of the sentence seems like the best choice